src/MinterGateway.sol

```diff
@@ -871,6 +876,20 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
         // MinterGateway already has more recent collateral update
         if (newTimestamp_ <= lastUpdateTimestamp_) revert StaleCollateralUpdate(newTimestamp_, lastUpdateTimestamp_);
 
+        uint32 maxUpdateCollateralFrequency_ = maxUpdateCollateralFrequency();
+
+        if (maxUpdateCollateralFrequency_ != 0) {
+            unchecked {
+                // NOTE: Can be done unchecked since above code already prevents `newTimestamp_ < lastUpdateTimestamp_`.
+                uint32 timeSinceLastUpdate_ = uint32(newTimestamp_ - lastUpdateTimestamp_);
+                uint32 minimumDelay_ = updateCollateralInterval() / maxUpdateCollateralFrequency_;
+
+                if (timeSinceLastUpdate_ < minimumDelay_) {
+                    revert CollateralUpdateTooSoon(timeSinceLastUpdate_, minimumDelay_);
```

Comment:
I would have named this error: `UpdateCollateralFrequencyExceeded`
```suggestion
                    revert UpdateCollateralFrequencyExceeded(timeSinceLastUpdate_, minimumDelay_);
```

Reply:
The issue is that, from a user standpoint, the frequency is but a TTG parameter, but instead the UX effect is that you cannot update too soon, regardless of the frequency being 5 or whatnot. So saying "frequency exceeeded" would be odd since the user might just think "but I only did it once time?".